include/psa/crypto_builtin_composites.h

```diff
@@ -112,4 +112,75 @@ typedef struct
 
 #define MBEDTLS_PSA_AEAD_OPERATION_INIT {0, 0, 0, 0, {0}}
 
+#include "mbedtls/ecdsa.h"
+
+/* Context structure for the Mbed TLS interruptible sign hash implementation. */
+typedef struct
+{
+#if ( defined(MBEDTLS_PSA_BUILTIN_ALG_ECDSA) || \
+ defined(MBEDTLS_PSA_BUILTIN_ALG_DETERMINISTIC_ECDSA) ) && \
+ defined(MBEDTLS_ECP_RESTARTABLE)
+    mbedtls_ecdsa_context * MBEDTLS_PRIVATE( ctx );
+    mbedtls_ecdsa_restart_ctx MBEDTLS_PRIVATE( restart_ctx );
+
+    size_t MBEDTLS_PRIVATE( coordinate_bytes );
+    psa_algorithm_t MBEDTLS_PRIVATE( alg );
+    mbedtls_md_type_t MBEDTLS_PRIVATE( md_alg );
+    uint8_t MBEDTLS_PRIVATE(hash)[PSA_BITS_TO_BYTES(PSA_VENDOR_ECC_MAX_CURVE_BITS)];
+    size_t MBEDTLS_PRIVATE( hash_length );
+
+#else
+    /* Make the struct non-empty if algs not supported. */
+    unsigned MBEDTLS_PRIVATE(dummy);
+
+#endif /* MBEDTLS_PSA_BUILTIN_ALG_ECDSA) ||
```

Comment:
Minor: leftover `)`. (Also, could use parentheses for operators: (a || b) && c.)